src/behavioural-events/events-tracker.ts

```diff
@@ -23,6 +27,12 @@ export interface IEventsTracker {
   trackCheckoutSessionStart(
     props: CheckoutSessionStartEventProps,
   ): Promise<void>;
+  trackBillingEmailEntryImpression(props: UserEventProps): Promise<void>;
+  trackBillingEmailEntrySubmit(props: UserEventProps): Promise<void>;
+  trackBillingEmailEntryDismiss(props: UserEventProps): Promise<void>;
+  trackBillingEmailEntryError(
+    props: BillingEmailEntryErrorEventProps,
+  ): Promise<void>;
```

Comment:
Hmm sorry I missed this on previous PRs but do these need to be part of the public API? I would try to avoid that as much as possible specially considering these seem to only be needed internally?

Reply:
Fortunately we haven't made a release with these new SDK methods, so I believe we're on time to remove them and avoid the breaking change.

Reply:
We don't! I fixed this in the upstream branch `web-1910` but I did not do the change on this one yet.

Reply:
No problem, thanks for reviewing!

But maybe wait for the reviews since I am still working on this draft PR and there may be larger changes. I will move to "Ready for review" when it's good to go from my side.